core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -174,9 +177,17 @@ public List<User> doListUsersWithID(String filter, int maxItemLimit) throws User
         String displayNameAttribute = realmConfig.getUserStoreProperty(JDBCUserStoreConstants.DISPLAY_NAME_ATTRIBUTE);
         String domain = realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
         try {
-
+            if (isCaseSensitiveUsername()) {
+                sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.GET_USER_FILTER_WITH_ID);
+            } else {
+                sqlStmt = realmConfig
+                        .getUserStoreProperty(JDBCCaseInsensitiveConstants.GET_USER_FILTER_WITH_ID_CASE_INSENSITIVE);
+            }
             if (filter != null && filter.trim().length() != 0) {
                 filter = filter.trim();
+                if (SQL_LIKE_SYNTAX_PATTERN.matcher(sqlStmt).find()) {
+                    filter = filter.replace("_", "\\_");
+                }
```

Comment:
The filter(username search queries with `_`) is changed only if the sql operation included `LIKE` operation. 
There can be situations where custom user store managers which can decide to have custom queries defined without `LIKE` operator hence adding escape character for the filter can cause issues.
Even there can be custom extension developed without sql queries (external API calls) to do the user management operations. In such cases also, the escape character should not be applied.